[2025-03-06T16:27:05.249+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-03-06T16:27:05.277+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:27:01.474767+00:00 [queued]>
[2025-03-06T16:27:05.289+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:27:01.474767+00:00 [queued]>
[2025-03-06T16:27:05.292+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-03-06T16:27:05.307+0000] {taskinstance.py:2890} INFO - Executing <Task(SparkSubmitOperator): load_exchangeRate_python_job> on 2025-03-06 16:27:01.474767+00:00
[2025-03-06T16:27:05.314+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'spark_flow', 'load_exchangeRate_python_job', 'manual__2025-03-06T16:27:01.474767+00:00', '--job-id', '10', '--raw', '--subdir', 'DAGS_FOLDER/dag_***_spark.py', '--cfg-path', '/tmp/tmp5mcpf8cq']
[2025-03-06T16:27:05.318+0000] {standard_task_runner.py:105} INFO - Job 10: Subtask load_exchangeRate_python_job
[2025-03-06T16:27:05.319+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=596) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-03-06T16:27:05.320+0000] {standard_task_runner.py:72} INFO - Started process 597 to run task
[2025-03-06T16:27:05.377+0000] {task_command.py:467} INFO - Running <TaskInstance: spark_flow.load_exchangeRate_python_job manual__2025-03-06T16:27:01.474767+00:00 [running]> on host bfcfbd2155da
[2025-03-06T16:27:05.500+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='haonv' AIRFLOW_CTX_DAG_ID='spark_flow' AIRFLOW_CTX_TASK_ID='load_exchangeRate_python_job' AIRFLOW_CTX_EXECUTION_DATE='2025-03-06T16:27:01.474767+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-06T16:27:01.474767+00:00'
[2025-03-06T16:27:05.501+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-03-06T16:27:05.530+0000] {base.py:84} INFO - Retrieving connection 'spark-conn'
[2025-03-06T16:27:05.532+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master spark://172.18.0.3:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py
[2025-03-06T16:27:08.576+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO SparkContext: Running Spark version 3.5.5
[2025-03-06T16:27:08.579+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO SparkContext: OS info Linux, 5.15.167.4-microsoft-standard-WSL2, amd64
[2025-03-06T16:27:08.579+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO SparkContext: Java version 17.0.14
[2025-03-06T16:27:08.628+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-03-06T16:27:08.702+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO ResourceUtils: ==============================================================
[2025-03-06T16:27:08.702+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-03-06T16:27:08.703+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO ResourceUtils: ==============================================================
[2025-03-06T16:27:08.704+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO SparkContext: Submitted application: ExchangeRateToDW
[2025-03-06T16:27:08.725+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-03-06T16:27:08.734+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO ResourceProfile: Limiting resource is cpu
[2025-03-06T16:27:08.735+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-03-06T16:27:08.777+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO SecurityManager: Changing view acls to: ***
[2025-03-06T16:27:08.778+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO SecurityManager: Changing modify acls to: ***
[2025-03-06T16:27:08.779+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO SecurityManager: Changing view acls groups to:
[2025-03-06T16:27:08.779+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO SecurityManager: Changing modify acls groups to:
[2025-03-06T16:27:08.780+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2025-03-06T16:27:08.998+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:08 INFO Utils: Successfully started service 'sparkDriver' on port 43497.
[2025-03-06T16:27:09.026+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO SparkEnv: Registering MapOutputTracker
[2025-03-06T16:27:09.061+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO SparkEnv: Registering BlockManagerMaster
[2025-03-06T16:27:09.077+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-03-06T16:27:09.077+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-03-06T16:27:09.081+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-03-06T16:27:09.105+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-3b92306a-d075-4985-90bc-047a5dbbe152
[2025-03-06T16:27:09.121+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-03-06T16:27:09.135+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-03-06T16:27:09.250+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-03-06T16:27:09.307+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-03-06T16:27:09.340+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 ERROR SparkContext: Failed to add /opt/bitnami/spark/jars/postgresql/postgresql-42.7.5.jar to Spark environment
[2025-03-06T16:27:09.341+0000] {spark_submit.py:641} INFO - java.io.FileNotFoundException: Jar /opt/bitnami/spark/jars/postgresql/postgresql-42.7.5.jar not found
[2025-03-06T16:27:09.342+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:2095)
[2025-03-06T16:27:09.342+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.addJar(SparkContext.scala:2151)
[2025-03-06T16:27:09.343+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.$anonfun$new$15(SparkContext.scala:521)
[2025-03-06T16:27:09.343+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.$anonfun$new$15$adapted(SparkContext.scala:521)
[2025-03-06T16:27:09.344+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-03-06T16:27:09.345+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-03-06T16:27:09.345+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-03-06T16:27:09.346+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:521)
[2025-03-06T16:27:09.346+0000] {spark_submit.py:641} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2025-03-06T16:27:09.347+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-03-06T16:27:09.347+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-03-06T16:27:09.348+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-03-06T16:27:09.348+0000] {spark_submit.py:641} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-03-06T16:27:09.349+0000] {spark_submit.py:641} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-03-06T16:27:09.349+0000] {spark_submit.py:641} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2025-03-06T16:27:09.350+0000] {spark_submit.py:641} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-03-06T16:27:09.350+0000] {spark_submit.py:641} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2025-03-06T16:27:09.351+0000] {spark_submit.py:641} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2025-03-06T16:27:09.351+0000] {spark_submit.py:641} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2025-03-06T16:27:09.352+0000] {spark_submit.py:641} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-03-06T16:27:09.353+0000] {spark_submit.py:641} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-03-06T16:27:09.353+0000] {spark_submit.py:641} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-06T16:27:09.427+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://172.18.0.3:7077...
[2025-03-06T16:27:09.467+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO TransportClientFactory: Successfully created connection to /172.18.0.3:7077 after 21 ms (0 ms spent in bootstraps)
[2025-03-06T16:27:09.546+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20250306162709-0002
[2025-03-06T16:27:09.549+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250306162709-0002/0 on worker-20250306162317-172.18.0.10-36791 (172.18.0.10:36791) with 2 core(s)
[2025-03-06T16:27:09.552+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO StandaloneSchedulerBackend: Granted executor ID app-20250306162709-0002/0 on hostPort 172.18.0.10:36791 with 2 core(s), 1024.0 MiB RAM
[2025-03-06T16:27:09.553+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20250306162709-0002/1 on worker-20250306162317-172.18.0.8-46753 (172.18.0.8:46753) with 2 core(s)
[2025-03-06T16:27:09.554+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO StandaloneSchedulerBackend: Granted executor ID app-20250306162709-0002/1 on hostPort 172.18.0.8:46753 with 2 core(s), 1024.0 MiB RAM
[2025-03-06T16:27:09.557+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 41515.
[2025-03-06T16:27:09.558+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO NettyBlockTransferService: Server created on bfcfbd2155da:41515
[2025-03-06T16:27:09.559+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-03-06T16:27:09.565+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, bfcfbd2155da, 41515, None)
[2025-03-06T16:27:09.570+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO BlockManagerMasterEndpoint: Registering block manager bfcfbd2155da:41515 with 434.4 MiB RAM, BlockManagerId(driver, bfcfbd2155da, 41515, None)
[2025-03-06T16:27:09.575+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, bfcfbd2155da, 41515, None)
[2025-03-06T16:27:09.578+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, bfcfbd2155da, 41515, None)
[2025-03-06T16:27:09.591+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250306162709-0002/0 is now RUNNING
[2025-03-06T16:27:09.595+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20250306162709-0002/1 is now RUNNING
[2025-03-06T16:27:09.887+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:09 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2025-03-06T16:27:10.360+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:10 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-03-06T16:27:10.373+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:10 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2025-03-06T16:27:12.638+0000] {spark_submit.py:641} INFO - Traceback (most recent call last):
[2025-03-06T16:27:12.640+0000] {spark_submit.py:641} INFO - File "/opt/***/jobs/python/load_exchangeRateToDW.py", line 66, in <module>
[2025-03-06T16:27:12.643+0000] {spark_submit.py:641} INFO - .load()
[2025-03-06T16:27:12.644+0000] {spark_submit.py:641} INFO - ^^^^^^
[2025-03-06T16:27:12.645+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/streaming/readwriter.py", line 304, in load
[2025-03-06T16:27:12.647+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-03-06T16:27:12.650+0000] {spark_submit.py:641} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 185, in deco
[2025-03-06T16:27:12.672+0000] {spark_submit.py:641} INFO - pyspark.errors.exceptions.captured.AnalysisException: Failed to find data source: kafka. Please deploy the application as per the deployment section of Structured Streaming + Kafka Integration Guide.
[2025-03-06T16:27:12.799+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO SparkContext: Invoking stop() from shutdown hook
[2025-03-06T16:27:12.802+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-03-06T16:27:12.818+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO SparkUI: Stopped Spark web UI at http://bfcfbd2155da:4040
[2025-03-06T16:27:12.824+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO StandaloneSchedulerBackend: Shutting down all executors
[2025-03-06T16:27:12.830+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2025-03-06T16:27:12.853+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-03-06T16:27:12.867+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO MemoryStore: MemoryStore cleared
[2025-03-06T16:27:12.867+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO BlockManager: BlockManager stopped
[2025-03-06T16:27:12.873+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-03-06T16:27:12.875+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-03-06T16:27:12.888+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO SparkContext: Successfully stopped SparkContext
[2025-03-06T16:27:12.889+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO ShutdownHookManager: Shutdown hook called
[2025-03-06T16:27:12.889+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO ShutdownHookManager: Deleting directory /tmp/spark-1cf4d121-2e66-423b-98b9-8df3b601f256/pyspark-4dc24cac-3fbd-4aae-b551-1ce8291b9fc4
[2025-03-06T16:27:12.893+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO ShutdownHookManager: Deleting directory /tmp/spark-e6e728bd-af40-488e-bc6c-2f1f05158b61
[2025-03-06T16:27:12.897+0000] {spark_submit.py:641} INFO - 25/03/06 16:27:12 INFO ShutdownHookManager: Deleting directory /tmp/spark-1cf4d121-2e66-423b-98b9-8df3b601f256
[2025-03-06T16:27:12.936+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://172.18.0.3:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.
[2025-03-06T16:27:12.940+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=spark_flow, task_id=load_exchangeRate_python_job, run_id=manual__2025-03-06T16:27:01.474767+00:00, execution_date=20250306T162701, start_date=20250306T162705, end_date=20250306T162712
[2025-03-06T16:27:12.975+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-03-06T16:27:12.976+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 10 for task load_exchangeRate_python_job (Cannot execute: spark-submit --master spark://172.18.0.3:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.; 597)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://172.18.0.3:7077 --name arrow-spark --deploy-mode client jobs/python/load_exchangeRateToDW.py. Error code is: 1.
[2025-03-06T16:27:13.006+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-03-06T16:27:13.034+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-03-06T16:27:13.040+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
